### ISSUE TITLE
NX Release: No changes were detected using git history and the conventional commits standard

### ISSUE THREAD

**Issue by nnelson-cen65665** [nrwl/nx#22251]
### Current Behavior

Hi, I'm trying to get to work with NX Release. Everything works, but it is not detecting changes in libs. I have two apps with package.json that I want to independently version. These two apps are built with 4-5 libs (they don't have package.json, and I don't want them to have versions). When I make some global changes (like configuration changes), NX Release bumps the version and creates tags. But when I do some changes in libs, NX Release says: 'No changes were detected using git history and the conventional commits standard.' What am I missing or doing wrong? Thx

"release": {
    "projects": ["app1", "app2"],
    "projectsRelationship": "independent",
    "releaseTagPattern": "{projectName}-v{version}",
    "changelog": {
      "projectChangelogs": true
    },
    "version": {
      "conventionalCommits": true
    }
  }

### Expected Behavior

When I change something in libs it will bump version of an app depends on it. 

### Steps to Reproduce

1. create and app in workspace
2. create an lib 
3. create some component or fn in the lib
4. use the component or fn in the app
5. push changes to git (nx release will work)
6. make changes to the lib code and commit
7. try nx release and it not bump the app version

### Nx Report

```shell
Node   : 16.15.1
OS     : darwin-arm64
npm    : 9.2.0

nx                 : 18.0.8
@nx/js             : 18.0.8
@nx/jest           : 18.0.8
@nx/linter         : 18.0.8
@nx/eslint         : 18.0.8
@nx/workspace      : 18.0.8
@nx/cypress        : 18.0.8
@nx/devkit         : 18.0.8
@nx/eslint-plugin  : 18.0.8
@nx/react          : 18.0.8
@nrwl/tao          : 18.0.8
@nx/web            : 18.0.8
@nx/webpack        : 18.0.8
typescript         : 5.3.2
```

### Failure Logs

```shell
app 🔍 Reading data for package "app" from apps/app/package.json
app 📄 Resolved the current version as 0.xx.3 from git tag "app-v0.xx.3".
app 🚫 No changes were detected using git history and the conventional commits standard.
app 🚫 Skipping versioning "app" as no changes were detected.
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by ryan-mcginty-alation** [comment]
Could be similar to this -> https://github.com/nrwl/nx/issues/22268

**Comment by andrei-ilyukovich** [comment]
Seems like I have the same or similar issue.
I have 'app' depends on 'lib'. I want both projects to be release. But it says `No changes were detected using git history and the conventional commits standard.` for both `app` and `lib` despite the fact I made a proper conventional commits to those projects like:
`fix(app): make a fix to app`
`fix(lib): make a fix to lib`

@nnelson-cen65665 do you have the same if you make a commit to your app directly and not to the dependency?

My config:
  ```
"release": {
    "projects": ["app", "lib"],
    "projectsRelationship": "independent",
    "changelog": {
      "projectChangelogs": true
    },
    "version": {
      "conventionalCommits": true
    }
  }
```

**Comment by nnelson-cen65665** [comment]
I think that changes to app works ok but any change to libs do not bump version of the app. For now I switched to @jscutlery/semver.

**Comment by fahslaj** [comment]
This was fixed on Nx 19.1.0 with this PR https://github.com/nrwl/nx/pull/23252
